packages/x-data-grid-premium/src/hooks/features/aggregation/createAggregationLookup.ts

```diff
@@ -25,13 +26,21 @@ const getAggregationCellValue = ({
   field,
   aggregationFunction,
   aggregationRowsScope,
+  isDataSource,
 }: {
-  apiRef: React.MutableRefObject<GridApiPremium>;
+  apiRef: React.MutableRefObject<GridPrivateApiPremium>;
   groupId: GridRowId;
   field: string;
-  aggregationFunction: GridAggregationFunction;
+  aggregationFunction: GridAggregationFunction | GridAggregationFunctionDataSource;
   aggregationRowsScope: DataGridPremiumProcessedProps['aggregationRowsScope'];
+  isDataSource: boolean;
 }) => {
+  if (isDataSource) {
+    return apiRef.current.internal_getAggregatedValue(groupId, field);
```

Comment:
why having `internal` prefix if it is in the private API?

Reply:
[Renamed](https://github.com/mui/mui-x/pull/15741/commits/89cc04d73e850b12a5ce6d97d37e4e3af63e043d) it to `resolveGroupAggregation`.